package/src/lsif.ts

```diff
@@ -0,0 +1,229 @@
+import * as sourcegraph from 'sourcegraph'
+import * as LSP from 'vscode-languageserver-types'
+import { convertLocations, convertHover } from './lsp-conversion'
+import { queryGraphQL } from './api'
+
+function repositoryFromDoc(doc: sourcegraph.TextDocument): string {
+    const url = new URL(doc.uri)
+    return url.hostname + url.pathname
+}
+
+function commitFromDoc(doc: sourcegraph.TextDocument): string {
+    const url = new URL(doc.uri)
+    return url.search.slice(1)
+}
+
+function pathFromDoc(doc: sourcegraph.TextDocument): string {
+    const url = new URL(doc.uri)
+    return url.hash.slice(1)
+}
+
+function setPath(doc: sourcegraph.TextDocument, path: string): string {
+    const url = new URL(doc.uri)
+    url.hash = path
+    return url.href
+}
+
+async function queryLSIF({
+    doc,
+    method,
+    path,
+    position,
+}: {
+    doc: sourcegraph.TextDocument
+    method: string
+    path: string
+    position: LSP.Position
+}): Promise<any> {
+    const url = new URL(
+        '.api/lsif/request',
+        sourcegraph.internal.sourcegraphURL
+    )
+    url.searchParams.set('repository', repositoryFromDoc(doc))
+    url.searchParams.set('commit', commitFromDoc(doc))
+
+    const response = await fetch(url.href, {
+        method: 'POST',
+        headers: new Headers({
+            'content-type': 'application/json',
+            'x-requested-with': 'Sourcegraph LSIF extension',
```

Comment:
This isn't accurate anymore, right?

Reply:
That's true, switching to `'Basic code intel'`